include/manif/impl/se3/SE3_base.h

```diff
@@ -206,6 +231,38 @@ SE3Base<_Derived>::translation() const
   return coeffs().template head<3>();
 }
 
+template <typename _Derived>
+void SE3Base<_Derived>::quat(const QuaternionDataType& quaternion)
+{
+  quat(quaternion.coeffs());
+}
+
+template <typename _Derived>
+template <typename _EigenDerived>
+void SE3Base<_Derived>::quat(const Eigen::MatrixBase<_EigenDerived>& quaternion)
+{
+  using std::abs;
+  assert_vector_dim(quaternion, 4);
+  MANIF_ASSERT(abs(quaternion.template norm()-Scalar(1)) <
+               Constants<Scalar>::eps_s,
+               "The quaternion is not normalized !",
+               invalid_argument);
+
+  asSO3().coeffs() = quaternion;
+}
+
+template <typename _Derived>
+void SE3Base<_Derived>::quat(const SO3<Scalar>& so3)
+{
+  quat(so3.quat());
```

Comment:
Last comment, lets avoid creating a temporary Quaternion object,
```suggestion
  quat(so3.coeffs());
```